examples/sunlinsol/ginkgo/test_sunlinsol_ginkgo.cpp

```diff
@@ -143,7 +148,45 @@ void fill_matrix(gko::matrix::Csr<sunrealtype, sunindextype>* matrix)
   unsigned num_blocks        = (mat_rows + threads_per_block - 1) / threads_per_block;
 
   fill_kernel<<<num_blocks, threads_per_block>>>(mat_rows, mat_cols, row_ptrs, col_idxs, mat_data);
-  HIP_OR_CUDA(hipDeviceSynchronize();, cudaDeviceSynchronize(););
+  HIP_OR_CUDA_OR_SYCL(hipDeviceSynchronize(), cudaDeviceSynchronize(), matrix->get_executor()->synchronize());
```

Comment:
This line would only compile when HIP or CUDA is used, so I think you can revert this line back.

Reply:
I revert them back, but I need to provide `HIP_OR_CUDA` and `HIP_OR_CUDA_OR_SYCL`.
which one do you prefer?

Reply:
I think we should actually go with
```
 HIP_OR_CUDA_OR_SYCL(hipDeviceSynchronize(), cudaDeviceSynchronize(), );
```
Then we only need one of the macros, but it is also clear that nothing happens in the SYCL case in this block of code.